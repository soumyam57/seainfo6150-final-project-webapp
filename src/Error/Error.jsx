import React from 'react';
import styles from "./Error.module.css";

const Error = () => {
    return (
        <div className={styles.container}>
            <div className={styles.main}>  
                <h1>OOPS!!!</h1>
                <h2>404</h2>
                <h4>I am sorry, the page you requested could not be found</h4>
                <a href = '/'> Go back to HomePage</a>
            </div>
        </div>
    )
}

export default Error;
