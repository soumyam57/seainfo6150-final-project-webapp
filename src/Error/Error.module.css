.container {
    width: 100%;
    height: 100%;
    position: absolute;
    text-align: center;
    color: rgb(53, 23, 23);
    background-color:rgb(131, 64, 64);
}
.main {
    padding-top: 5%;
}

.container h2 {
    padding: 1%;
    font-size: 1500%;
    font-family: 'Space Grotesk', sans-serif;
}
.container h4 {
    font-size: 200%;
    padding: 20px;
    font-family: 'Space Grotesk', sans-serif;
}

.container a{  
    border: 2px solid rgb(153, 127, 127);
    text-decoration: none;
    background : rgba(204, 0, 0, 0.5);
    color: #fff;
    padding :12px 24px;
    display: inline-block;
    border-radius: 25px;
    font-size: 14px;
    text-transform: uppercase;
    transition: 0.4s;
    font-family: 'Space Grotesk', sans-serif;
}
.container a:hover{
    background:rgb(83, 18, 18);
    font-family: 'Space Grotesk', sans-serif;
}

