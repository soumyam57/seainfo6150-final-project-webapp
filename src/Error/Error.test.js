import React from "react";
import { render } from "@testing-library/react";
import Error from "./Error.jsx";
import { BrowserRouter } from 'react-router-dom';

it("renders the Error component correctly", () => {
  const { container } = render(
    <BrowserRouter>  
        <Error/>
    </BrowserRouter> 
  );
  expect(container).toMatchSnapshot();
});