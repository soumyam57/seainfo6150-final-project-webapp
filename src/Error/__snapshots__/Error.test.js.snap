// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Error component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="main"
    >
      <h1>
        OOPS!!!
      </h1>
      <h2>
        404
      </h2>
      <h4>
        I am sorry, the page you requested could not be found
      </h4>
      <a
        class="link"
        href="/"
      >
        Go back to Homepage
      </a>
    </div>
  </div>
</div>
`;
